### PR TITLE
Add resnet50 example

### PR DESCRIPTION
**Pull request by keisukefukuda**
* Added ResNet50 example, originally written by Kudo-san
* Modified train_imagenet.py accordingly.

The result is:

    epoch       iteration   main/loss   validation/main/loss  main/accuracy  validation/main/accuracy  lr
    2           1000        1.79846                           0.377313                                 0.01
    4           2000        1.28585                           0.525031                                 0.01
    7           3000        1.13306                           0.581438                                 0.01
    9           4000        1.02884                           0.61925                                  0.01

### CODE REVIEW

File: examples/imagenet/resnet50.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+    def __call__(self, x):
+        h = F.relu(self.bn1(self.conv1(x)))
+        h = F.relu(self.bn2(self.conv2(h)))
+        h = self.bn3(self.conv3(h))
+
+        return F.relu(h + x)
+
+
+class Block(chainer.Chain):
```
Comment (beam2d) [inline examples/imagenet/resnet50.py]: I feel it's better to inherit `ChainList` instead of `Chain`, and just pass each bottleneck child link to `add_link` one by one in `__init__`. Then, the `__call__` is also made simple.

File: examples/imagenet/resnet50.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+                                  initialW=initializers.HeNormal(),
+                                  nobias=True),
+            bn1=L.BatchNormalization(64),
+            res2=Block(3, 64, 64, 256, 1),
+            res3=Block(4, 256, 128, 512),
+            res4=Block(6, 512, 256, 1024),
+            res5=Block(3, 1024, 512, 2048),
+            fc=L.Linear(2048, 1000),
+        )
+        self.train = True
```
Comment (beam2d) [inline examples/imagenet/resnet50.py]: Remove this unused attribute.

File: examples/imagenet/resnet50.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+    def clear(self):
+        self.loss = None
+        self.accuracy = None
+
+    def __call__(self, x, t):
+        self.clear()
```
Comment (beam2d) [inline examples/imagenet/resnet50.py]: Remove this line because `self.loss` and `self.accuracy` are not used. The `clear` method itself can also be removed.

File: examples/imagenet/resnet50.py (hunk 1 above)
Comment (mitmul) [inline examples/imagenet/resnet50.py]: @keisukefukuda This tutorial might be informative to know how to use `ChainList` for this case: http://docs.chainer.org/en/latest/tutorial/convnet.html#resnet152

**Comment by keisukefukuda** [comment]
I reflected the comments from @beam2d -san and @mitmul -san!

**Comment by beam2d** [comment]
LGTM. It runs in my environment. @mitmul Could you confirm the fix and merge it if it's ok?

File: examples/imagenet/resnet50.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+class ResNet50(chainer.Chain):
+
+    insize = 224
+
+    def __init__(self):
+        super(ResNet50, self).__init__(
+            conv1=L.Convolution2D(3, 64, 7, 2, 3,
+                                  initialW=initializers.HeNormal(),
+                                  nobias=True),
```
Comment (mitmul) [inline examples/imagenet/resnet50.py]: I don't know why (and feel it's strange) but the first convolution has bias term in the original implementation. See the original prototext here: https://github.com/KaimingHe/deep-residual-networks/blob/master/prototxt/ResNet-50-deploy.prototxt

Comment (keisukefukuda) [inline examples/imagenet/resnet50.py]: Thank you for the advice. I fixed it!

**Comment by mitmul** [comment]
Somehow the last commit wasn't merged (why? it's creepy...) so I submit a PR to fix that #2655
